src/node_modules/ringbuffer.js

```diff
@@ -0,0 +1,25 @@
+import {EventEmitter} from 'events';
+
+export class RingBuffer extends EventEmitter {
+	constructor(opts) {
+		super();
+		this.records = [];
+		this.limit   = opts.bufferSize;
+	}
+
+	write(log) {
+		this.records.push(log);
+		if (this.records.length > this.limit) {
```

Comment:
Ah, I missed that event. A single event will do. The only bit that I see missing then is document that it exists.  I would have the docs also recommend that developers use the event as a best practice.